Fix bug with one-time bulk payment via Stripe

nospame opened the pull request:
## Product Description
Fixes two related bugs where users could not make a bulk payment on the Billing Statements page with an unsaved card:
- when the customer had no saved cards, they would see a "Your card number is incorrect" error
- when the customer had any saved cards, they would see "Processing your payment..." forever

## Technical Summary
https://dimagi.atlassian.net/browse/SAAS-19009
As noted in the existing javascript comment, `self.showOrHideStripeUI` is called several times on initializing the page... and more for each card the customer has saved. The `cardElement` is mounted and unmounted a few times during this process - while this ultimately gets to the correct visual appearance, Stripe doesn't actually allow re-using an element for data collection like this, which results in `IntegrationError: We could not retrieve data from the specified Element.` when attempting to pay with new card details.

This resolves that issue by creating a new `cardElement` and mounting it each time we reset the stripe card UI - and only doing so when there is a matching DOM element to mount, so as to try to avoid creating cardElements that get immediately thrown away. There is also a change to also avoid initializing Stripe more than once during this process, so we're not adding many new API calls on every page load. A more comprehensive solution here _would_ potentially include not attempting to update/reset the stripe card UI so many times on a page load, but I think that's out of scope for this bugfix.

This is small by volume. It can probably be reviewed as files changed.

## Safety Assurance

### Safety story
Tested locally and on staging. Because stripe is used in many places, also confirmed payment on the Current Subscription and enterprise Billing Statements pages still work using freshly entered card details or a saved card, in situations where the customer has and does not have any cards already saved.

### Automated test coverage
Likely not for this javascript component.

### QA Plan
Not planning it, but could see an argument if reviewers think necessary.


### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

nospame committed:
Run cardElementPromise each time before mounting/unmounting cardElement

Ensures cardElement is fresh each time; Stripe does not allow re-use for data submission

nospame committed:
Only load stripe once

nospame committed:
Just call cardElementPromise on reset

No need to clear the element because we're not trying to reuse it anymore

nospame committed:
Don't explicitly mount/unmount elements

We're creating a new element every time now, so no need to call unmount or track the mounted status

nospame committed:
showOrHideStripeUI -> resetStripeCardUI

Also fold in cardElementPromise because it's simple and not used elsewhere

nospame commented on corehq/apps/accounting/static/accounting/js/payment_method_handler.js:
```diff
@@ -234,11 +229,7 @@ var paymentMethodHandler = function (formId, opts) {
+        self.resetStripeCardUI();
     };
 
     self.processPayment = function () {
```
`cardElement.clear()` is no longer relevant since we are not re-using the same `cardElement`, instead attaching a new one each time we show the Stripe Card UI. You're correct that this call to `resetStripeCardUI()` does nothing and can be removed.

nospame committed:
Remove useless call to resetStripeCardUI

MartinRiese reviewed (approved):
Just summarizing so I did not miss anything: We should no reuse the promise from `getCardElementPromise(initialPageData.get("stripe_public_key"))` so we are moving it into the function call. That makes `cardElementMounted` redundant and we can remove it.

nospame committed:
Unmount existing cardElement before creating a new one

Not strictly necessary but seems preferred practice with stripe.js

nospame committed:
Reset stripe card UI for real with self.reset

Fixes 'Your card number is incorrect' bug

nospame committed:
Rework resetStripeCardUI

- rename show -> shouldCreateElement
- only create an element if there is also a DOM element to mount

nospame committed:
Use arrow function
